### ISSUE TITLE
Build backend logic for welcome user component 

### ISSUE THREAD

**Issue by code2244** [Caleb-Cohen/Together#247]
## Please provide a summary of the feature

Adds a param user object to needsToBeWelcome. On the frontend if the current user has a needsToBeWelcomed, you show the modal.

When the user closes the modal, makes a request to a handler that updates the currently logged in user and removes the needsToBeWelcomed property/sets it to false.

## Are there any potential issues you foresee with this feature

The reusable modal feature needs to be merge into development before we can use a modal. So for the time being we will use a component.

## Is this feature related to any current or past issues? Please provide the #

Closes #203. Do to changing scope
Uses modal logic from #182 

## Which Branch should this fix be Pulled into?
PR into [development]

*If you're interested in working on this feature, please comment so you can be assigned*

**Comment by code2244** [comment]
I will like to work in this issue.

**Comment by code2244** [comment]
Hi everyone, so for this issue, with the help of Joe I tried to set the property needsToBeWelcome when the user log in but passport is giving some problems, in the passport middleware if the callback returns a successful user the session with the parameter does not get set but if the callback returns false the session get set.

`return cb(null, user);
 // return cb(null, false);`

for reference to this error here is the link for the branch I am using.
https://github.com/code2244/Together/tree/WelcomeUserComponentModal
